modules/sr/robot/connector.py

```diff
@@ -0,0 +1,30 @@
+from controller import Robot, Connector as WebotsConnector
+from sr.robot.utils import get_robot_device
+
+
+class Connector:
+    def __init__(self, webot: Robot):
+        self._connector = get_robot_device(webot, "Crane Connector", WebotsConnector)
+        self._connector.enablePresence(int(webot.getBasicTimeStep()))
+
+    @property
+    def lock(self) -> bool:
+        """
+        Get the current lock state of the connector. This does not indicate
+        whether a physical connection has been successfully made.
+        """
+        return self._connector.isLocked()
+
+    @lock.setter
+    def lock(self, value: bool) -> None:
```

Comment:
Any reason to use a setter rather than methods?

`lock()`
`unlock()`
`is_locked()`

Reply:
Mostly because we are now using properties for motors and all arduino interfaces